src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSparkUnitTest.java

```diff
@@ -0,0 +1,287 @@
+            final Tuple2<Long, URI> breakpoint = RunSGAViaProcessBuilderOnSpark.assignFASTQToBreakpoints(String.valueOf(i) + "\t" + Paths.get(fastqFileName).toUri());
+            rawFASTQFiles.add(breakpoint);
+
+            final String contigFileName = fastqFileName.replace("fastq", "pp.ec.filter.pass.rmdup.merged-contigs.fa");
+            expectedAssembledFASTAFiles.put(i++, Paths.get(contigFileName).toUri());
+        }
+    }
+
+
+    // parameter block
+    private static final int editDistanceTolerance = 2; // tolerance of the summed edit distance between expected contigs and corresponding actual assembled contigs
+
+    private static final int threads = 1;
+
+    @BeforeMethod
+    public void checkPATHTOSGAAvailability(){
+        if(null==sgaPath){
+            throw new SkipException("Skipping test because \"-DPath.To.SGA\" is set neither in system property nor as an environment variable.");
+        }
+    }
+
+    @Test(groups = "sv")
+    public void assemblyOneStepTest() throws IOException, InterruptedException, RuntimeException{
+
+        for(final Tuple2<Long, URI> breakpoint : rawFASTQFiles){
+
+            final RunSGAViaProcessBuilderOnSpark.SGAAssemblyResult result = RunSGAViaProcessBuilderOnSpark.performAssembly(breakpoint, sgaPath, threads, true)._2();
+
+            final File expectedAssembledContigFile = new File( expectedAssembledFASTAFiles.get(breakpoint._1()) );
+            final RunSGAViaProcessBuilderOnSpark.ContigsCollection expectedContigsCollection = new RunSGAViaProcessBuilderOnSpark.ContigsCollection(expectedAssembledContigFile);
+
+            Assert.assertTrue( compareContigs(result.assembledContigs, expectedContigsCollection) );
+        }
+    }
+
+    @Test(groups = "sv")
+    public void assemblyStepByStepTest() throws IOException, InterruptedException, RuntimeException{
+
+        for(final Tuple2<Long, URI> breakpoint : rawFASTQFiles){
+
+            // make temp dir, copy file
+            final File workingDir = Files.createTempDirectory("whatever").toAbsolutePath().toFile();
+            workingDir.deleteOnExit();
+
+            final String pathStringToRawFASTQ = breakpoint._2().getPath();
+            final String rawFASTQFileName = FilenameUtils.getName(pathStringToRawFASTQ);
+            BucketUtils.copyFile(pathStringToRawFASTQ, null, workingDir.toPath().toAbsolutePath().toString()+"/"+rawFASTQFileName);
+
+            final File fastqFile = new File(workingDir, FilenameUtils.getName(breakpoint._2().toString()));
+            final File expectedAssembledContigFile = new File( expectedAssembledFASTAFiles.get(breakpoint._1()) );
+
+            stepByStepTestWorker(workingDir, fastqFile, expectedAssembledContigFile);
+        }
+    }
+
+    private static void stepByStepTestWorker(final File workingDir, final File rawFASTQ, final File expectedAssembledContigFile) throws IOException, InterruptedException, RuntimeException{
```

Comment:
I'm not sure what value having this code in a separate method provides. Why not just inline it into the test method itself?